src/schema/generateFroidSchema.ts

```diff
@@ -301,10 +301,13 @@ export function generateFroidSchema(
   const subgraphs = [...currentSchemaMap.values()].map((sdl) => parse(sdl));
 
   // extract all definition nodes for federated schema
-  const allDefinitionNodes = subgraphs.reduce<DefinitionNode[]>(
+  let allDefinitionNodes = subgraphs.reduce<DefinitionNode[]>(
     (accumulator, value) => accumulator.concat(value.definitions),
     []
   );
+
+  allDefinitionNodes = removeInterfaceObjects(allDefinitionNodes);
+
```

Comment:
This is the main contribution I made to this file.  

The other changes were just refactoring to remove unnecessary variables

Reply:
nit: I would keep these variables immutable to prevent the variable from accidentally getting re-assigned/mutated when we don't want it to be. After we remove the interfaceObjects it isn't really "all definition nodes", so we can do something like:

```suggestion
  const allDefinitionNodes = subgraphs.reduce<DefinitionNode[]>(
    (accumulator, value) => accumulator.concat(value.definitions),
    []
  );

  const filteredDefinitionNodes = removeInterfaceObjects(allDefinitionNodes);

```

Reply:
good call, `allDefinitionNodes` is used later on, so ill make this change and make sure to rename the references to it later